Client/Example.cs

```
@@ -464,9 +498,8 @@ await foreach (var uiMessageEvent in uiMessageEventStream.ReadAllAsync())
 			}
 			catch (RpcException rpcException)
 			{
-				// When disconnecting from server, the client cancels the event stream. So, a cancelled
-				// exception should not be treated as an error.
-				if (rpcException.StatusCode == StatusCode.Cancelled)
+				// When disconnecting from the server, the client cancels the event stream. This usually results in an exception with a status code of StatusCode.Cancelled, but other codes are possible.
+				if (cancellationToken.IsCancellationRequested)
```

This is inside a general exception and thus this condition could hide the exception if the cancellation is also requested. Could you instead catch OperationCanceledException? Does this code throw such exception when cancelling?